examples/keras_recipes/sklearn_metric_callbacks.py

```diff
@@ -0,0 +1,142 @@
+TensorBoard using the `tf.summary` API.
+
+This template can be modified slightly to make it work with any existing sklearn metric.
+"""
+
+import tensorflow as tf
+from sklearn.metrics import jaccard_score
+import numpy as np
+
+
+class JaccardScoreCallback(tf.keras.callbacks.Callback):
+    """Computes the jaccard score and logs the results to TensorBoard."""
+
+    def __init__(self, model_fn, x_test, y_test, summary_writer):
+        self.model_fn = model_fn
+        self.x_test = x_test
+        self.y_test = y_test
+        self.summary_writer = summary_writer
+        self.keras_metric = tf.keras.metrics.Mean("jaccard_score")
+        self.epoch = 0
+
+    def on_epoch_end(self, batch, logs=None):
+        self.epoch += 1
+        self.keras_metric.reset_state()
+        predictions = self.model_fn(self.x_test)
+        jaccard_value = jaccard_score(
+            np.argmax(predictions, axis=-1), self.y_test, average=None
+        )
+        self.keras_metric.update_state(jaccard_value)
+        self._write_metric()
+
+    def _write_metric(self):
+        with self.summary_writer.as_default():
+            tf.summary.scalar(
+                self.keras_metric.name,
+                self.keras_metric.result().numpy().astype(float),
+                step=self.epoch,
+            )
+            self.summary_writer.flush()
+
+
+"""
+## Sample Usage
+Let's test our `JaccardScoreCallback` class with a real keras model.
+"""
+
+import tensorflow.keras as keras
+
+# Model / data parameters
+num_classes = 10
+input_shape = (28, 28, 1)
+
+# the data, split between train and test sets
+(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
+
+# Scale images to the [0, 1] range
+x_train = x_train.astype("float32") / 255
+x_test = x_test.astype("float32") / 255
+# Make sure images have shape (28, 28, 1)
+x_train = np.expand_dims(x_train, -1)
+x_test = np.expand_dims(x_test, -1)
+print("x_train shape:", x_train.shape)
+print(x_train.shape[0], "train samples")
+print(x_test.shape[0], "test samples")
+
+
+# convert class vectors to binary class matrices
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+
+import tensorflow.keras.layers as layers
+
+model = keras.Sequential(
+    [
+        keras.Input(shape=input_shape),
+        layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
+        layers.MaxPooling2D(pool_size=(2, 2)),
+        layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
+        layers.MaxPooling2D(pool_size=(2, 2)),
+        layers.Flatten(),
+        layers.Dropout(0.5),
+        layers.Dense(num_classes, activation="softmax"),
+    ]
+)
+
+model.summary()
+
+batch_size = 128
+epochs = 15
+
+model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
+summary_writer = tf.summary.create_file_writer("logs/traditional_classifier")
```

Comment:
I wonder if this should be moved to the callback constructor, so that all references to the summary API are within the callback class

Reply:
Done - yeah this is very nice and matches the tensorboard callback!